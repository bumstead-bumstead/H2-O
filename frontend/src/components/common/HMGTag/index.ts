export { default as HMGTag } from './HMGTag';
