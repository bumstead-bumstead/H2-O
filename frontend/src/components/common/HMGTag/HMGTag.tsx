import type { HTMLAttributes } from 'react';
import styled from '@emotion/styled';
import { Flex } from '@/components/common';

type TagVariant = 'default' | 'small';

interface Props extends HTMLAttributes<HTMLDivElement> {
  variant?: TagVariant;
}

function HMGTag({ variant = 'default', ...restProps }: Props) {
  return (
    <Container alignItems='center' justifyContent='center' variant={variant} {...restProps}>
      HMG Data
    </Container>
  );
}

export default HMGTag;

const Container = styled(Flex)<Props>`
  ${({ theme }) => theme.typography.HeadENBold10}
  background-color: ${({ theme }) => theme.colors.activeBlue2};
  color: ${({ theme }) => theme.colors.white};
  width: 70px;
  height: ${({ variant }) => (variant === 'small' ? 20 : 32)}px;
`;
