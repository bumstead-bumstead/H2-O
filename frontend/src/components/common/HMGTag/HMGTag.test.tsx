import HMGTag from './HMGTag';
import { render, screen, userEvent, waitFor } from '@/tests/test-util';

describe('HMGTag 컴포넌트 테스트', () => {
  it('HMGTag가 정상적으로 렌더링된다.', () => {
    render(<HMGTag />);

    expect(screen.getByText(/HMG Data/)).toBeInTheDocument();
  });

  it('onClick이 정상적으로 작동한다.', async () => {
    const mockFn = vi.fn();
    render(<HMGTag onClick={mockFn} />);

    userEvent.click(screen.getByText(/HMG Data/));

    await waitFor(() => expect(mockFn).toBeCalledTimes(1));
  });

  describe('HMGTag 스냅샷 테스트', () => {
    it('variant가 default일 때', () => {
      render(<HMGTag variant='default' />);

      expect(screen.getByText(/HMG Data/)).toMatchSnapshot();
    });

    it('variant가 small일 때', () => {
      render(<HMGTag variant='small' />);

      expect(screen.getByText(/HMG Data/)).toMatchSnapshot();
    });
  });
});
