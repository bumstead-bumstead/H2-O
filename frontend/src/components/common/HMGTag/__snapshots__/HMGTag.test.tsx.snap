// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HMGTag 컴포넌트 테스트 > HMGTag 스냅샷 테스트 > variant가 default일 때 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Hyundai Sans Head,sans-serif;
  font-size: 10px;
  font-weight: 700;
  line-height: 12px;
  letter-spacing: -0.01em;
  background-color: #00C3F0;
  color: #FFFFFF;
  width: 70px;
  height: 32px;
}

<div
  class="emotion-0"
>
  HMG Data
</div>
`;

exports[`HMGTag 컴포넌트 테스트 > HMGTag 스냅샷 테스트 > variant가 small일 때 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Hyundai Sans Head,sans-serif;
  font-size: 10px;
  font-weight: 700;
  line-height: 12px;
  letter-spacing: -0.01em;
  background-color: #00C3F0;
  color: #FFFFFF;
  width: 70px;
  height: 20px;
}

<div
  class="emotion-0"
>
  HMG Data
</div>
`;
